### ISSUE TITLE
`display-resample` behaves abnormally

### ISSUE THREAD

**Issue by inflation** [lhc70000/iina#741]
**System and IINA version:**

- macOS 10.13
- IINA 0.0.11

**Expected behavior:**

After setting
```
video-sync=display-resample
tscale=oversample
display-fps=60
```

IINA and mpv should behave the same.

**Actual behavior:**
IINA:
<img width="1072" alt="screen shot 2017-06-11 at 1 09 07 am" src="https://user-images.githubusercontent.com/2375962/27009427-238b1576-4e43-11e7-8892-42838e9c265a.png">
mpv:
<img width="1624" alt="screen shot 2017-06-11 at 1 09 52 am" src="https://user-images.githubusercontent.com/2375962/27009428-24bef430-4e43-11e7-8138-8d13de9629dc.png">

The actual refresh rate are extremely high in IINA.

**Steps to reproduce:**
Add above command to  `~/.config/mpv/mpv.conf` and use `stats.lua` plugin to see the result.

- [x] MPV does not have this problem.

**How often does this happen?**
All the time.

**Comment by Akemi** [comment]
i will just add some more info to this. i built iina just now, also with updated libmpv and libs. the issue is still there. i also use the opengl-cb API in the recently added [cocoa-cb backend](https://github.com/mpv-player/mpv/commit/c5e4538bc4b63f02bf89f62aa25a37b9eb0c0316) in mpv and there i have no such problems. since i limit the output to the display fps, i deactivated this mechanism for testing and it still estimates the display fps properly. also fixing #1542 sadly won't fix this problem either.

i also had a similar, maybe the same, problem when i tried to use a layer backed view with a normal NSOpenGLContext. the estimated display fps were just skyrocketing. though since you use a CAOpenGLLayer too, that shouldn't be the cause. maybe it's because of one of the overlaid views?

**Comment by Akemi** [comment]
i was able to reproduce the same problems problems in mpv when using a similar rendering loop too iina. i have to take back my previous comment and fixing #1542 indeed will also fix this problem and #125. just need to retrieve the proper display fps and set the property `display-fps` accordingly to the proper display fps.

**Comment by inflation** [comment]
Confirm fixed in `develop`
